core/spec/lib/i18n_spec.rb

```diff
@@ -19,12 +19,30 @@
   end
 
   describe '#available_locales' do
-    it 'returns locales defined in Spree and all other installed gems' do
-      locales = Spree.available_locales
+    context 'when SpreeI18n is defined' do
+      before do
+        class_double('SpreeI18n').
+          as_stubbed_const(transfer_nested_constants: true)
+        class_double('SpreeI18n::Locale', all: [:en, :de, :nl]).as_stubbed_const(transfer_nested_constants: true)
+      end
+
+      it 'returns all locales from the SpreeI18n' do
+        locales = Spree.available_locales
+
+        expected_locales = (['en', 'de', 'nl'] + I18n.available_locales).uniq.compact
+
+        expect(locales).to eq expected_locales
+      end
+    end
 
-      expected_locales = ([Rails.application.config.i18n.default_locale, I18n.locale, :en] + I18n.available_locales).uniq.compact
+    context 'when SpreeI18n is not defined' do
+      it 'returns just default locales' do
+        locales = Spree.available_locales
 
-      expect(locales).to eq expected_locales
+        expected_locales = ([Rails.application.config.i18n.default_locale, I18n.locale, :en] + I18n.available_locales).uniq.compact
```

Comment:
@MatthewKennedy this test will be flaky and will not show us that we have duplicates, it's better to avoid dynamic values for expect matcher and have a pre-defined static value :)